File: ruby/spec/enum.spec.rb
require 'enotype'

describe 'enum' do
  context 'with \'true\'' do
    it 'produces the expected result' do
      expect(Enotype::enum('true')).to be(true)
    end
  end
  
  context 'with \'false\'' do
    it 'produces the expected result' do
      expect(Enotype::enum('false')).to be(false)
    end
  end
  
  context 'with \'yes\'' do
    it 'produces the expected result' do
      expect(Enotype::enum('yes')).to be(true)
    end
  end
  
  context 'with \'no\'' do
    it 'produces the expected result' do
      expect(Enotype::enum('no')).to be(false)
    end
  end
  
  context 'with \'nope\'' do
    it 'raises an error' do
      expect { Enotype::enum('nope') }.to raise_error(RuntimeError)
    end
  end
end

File: php/spec/enum.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('enum', function() {
  describe('with \'true\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::enum('true'))->toBe(true);
    });
  });
  
  describe('with \'false\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::enum('false'))->toBe(false);
    });
  });
  
  describe('with \'yes\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::enum('yes'))->toBe(true);
    });
  });
  
  describe('with \'no\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::enum('no'))->toBe(false);
    });
  });
  
  describe('with \'nope\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::enum('nope'); })->toThrow();
    });
  });
});